vars/gangplank.groovy

```diff
@@ -0,0 +1,143 @@
+// Run gangplank
+
+// Available parameters:
+//     artifacts:    list    -- list of artifacts to be built
+//     extraFlags    string  -- Extra flags to use
+//     image:        string  -- Name of the image to be used
+//     mode:         string  -- Gangplank Mode
+//     workDir:      string  -- cosa working directory
+
+def buildArtifact(params = [:]) {
+    gangplankCmd = getMode(params)
+    gangplankCmd = getArtifacts(params, gangplankCmd)
+    runGangplank(gangplankCmd)
+}
+
+// Available parameters:
+//     artifacts:    list    -- list of artifacts to be built
+//     extraFlags    string  -- Extra flags to use
+//     image:        string  -- Name of the image to be used
+//     mode:         string  -- Gangplank Mode
+//     minioServeDir string  -- location to service minio from
+//     workDir:      string  -- cosa working directory
+
+def buildParallelArtifacts(params = [:]) {
+    gangplankCmd = getMode(params)
+    if (params['artifacts']) {
+        def artifacts = params['artifacts']
+        def configFile = startMinio(params)
+
+        // Call one pod for each artifact in parallel
+        parallel artifacts.inject([:]) { d, i -> d[i] = {
+            startParallel(configFile, "${gangplankCmd} -A ${i} ")
+        }; d }
+        finalize()
+    }
+}
+
+// Available parameters:
+//     artifacts:    list    -- list of artifacts to be built
+//     extraFlags    string  -- Extra flags to use
+//     singlePod     boolean -- Run generateSinglePod
+
+def generateSpec(params = [:]) {
+    if(params['singlePod']) {
+        gangplankCmd = "gangplank generateSinglePod  "
+    } else {
+        gangplankCmd = "gangplank generate "
+    }
+    gangplankCmd = getArtifacts(params, gangplankCmd)
+    gangplankCmd = getFlags(params, gangplankCmd)
+    fileName = "/tmp/jobspec-${UUID.randomUUID()}.spec"
+    gangplankCmd += "--yaml-out ${fileName} "
+    runGangplank(gangplankCmd)
+    return fileName
+}
+
+def startParallel(configFile, gangplankCmd) {
+    gangplankCmd += "-m ${configFile}"
+    runGangplank(gangplankCmd)
+}
+
+def startMinio(params =[:]) {
```

Comment:
I think basically every function which takes the generic `params` arg and is intended to be "user-facing" should have a docstring with supported params. We can add one to this one in a follow-up! :)